hal/stm32f373/spim.c

```diff
@@ -240,18 +268,18 @@ void spim_xfer(spim_t *spim, spim_xfer_opts *opts, uint16_t addr, void *read_buf
 	}
 	else
 	{
-		// since we complete when the last byte is read (this obviously happens after the last byte
-		// is transmitted) we don't need a complete routine for this.
-		spim->tx_dma_req.complete = NULL;
-		spim->tx_dma_req.complete_param = NULL;
+		spim->tx_dma_req.complete = spim_tx_dma_complete;
```

Comment:
There is now a callback for both tx and rx complete, and those check to see if the other one is complete before calling the original callback. That's not as efficient as before, so it would be nice to find a more efficient solution. The problem before was that you could not start another SPI transfer when the rx complete callback was called because the DMA tx complete had not yet been called and dma->reqs was not null (so the new request could not be started).
You would expect the tx to fire first, but occasionally the DMA tx complete interrupt runs after the DMA rx complete.